Canary docs (#5363)
solo-io/gloo: docs/content/operations/upgrading/1.9.0.md

---
title: 1.9.0+ Upgrade Notice
weight: 5
description: Migrating to Gloo Edge 1.9.x and Gloo Edge Enterprise 1.9.x
---

> In this guide we will describe the necessary steps to upgrade your Gloo Edge or Gloo Edge Enterprise deployments to their `1.9`
> versions. We recommend that you follow these steps only after you have followed our [guide to upgrade to 1.8]({{< versioned_link_path fromRoot="/operations/upgrading/1.8.0" >}}).

Reviewer: Some rephrasing suggestions...
```suggestion
Follow these steps to upgrade your Gloo Edge or Gloo Edge Enterprise deployments to version 1.9 from version 1.8. If your Gloo Edge deployment runs a version older than 1.8, make sure to [upgrade to 1.8]({{< versioned_link_path fromRoot="/operations/upgrading/1.8.0" >}}) before upgrading to 1.9.
```

> This upgrade guide also assumes that was gloo installed via `helm` or with `glooctl` version 1.8.0+
> (i.e., gloo is a helm release named "gloo", which you can confirm exists by running `helm ls --all-namespaces`).

Reviewer: ```suggestion
This upgrade guide assumes that you installed Gloo Edge with Helm or `glooctl`. You can verify this installation by checking for a Helm chart release named `gloo` in the output of `helm ls --all-namespaces`.
```

> Also, please make sure to check out our [general configuration recommendations]({{< versioned_link_path fromRoot="/operations/upgrading/upgrade_steps#upgrading-the-server-components" >}}) to avoid downtime during upgrades.

Reviewer: ```suggestion
For steps to avoid downtime during upgrades, check out the [Recommended settings]({{< versioned_link_path fromRoot="/operations/upgrading/upgrade_steps#upgrading-the-server-components" >}}).
```

> ##### CRD field updates

Reviewer: I would make these H2s unless there's a reason for H5.
```suggestion
## CRD field updates
```

> Any new fields added to our CRDs (e.g. added between 1.8.0 -> 1.9.0) need to be added to our validation schemas.

Reviewer: I noticed the 1.8 section above said between `1.8.0 -> 1.8.8`, like patch versions within the minor version, so just wanted to verify this should be different and include across minor versions (1.8.0 -> 1.9.0)

Author: it can happen either way, new fields have been added in patch versions (regretfully, as this is not strict semver) as well as minor versions

To apply the new CRDs:
```
helm pull gloo/gloo --version $GLOO_VERSION --untar #e.g. GLOO_VERSION=1.9.0
kubectl apply -f gloo/crds
```

{{% notice note %}}

> To support k8s 1.22, Gloo Edge CRDs are now written as v1 rather than v1beta1. CRDs that have been persisted as
> v1beta1 can be read using the v1 API in k8s 1.22 and are converted on read to the new API.

Reviewer: ```suggestion
To support Kubernetes 1.22, Gloo Edge CRDs are now written with the `v1` rather than the `v1beta1` API. CRDs that persisted as `v1beta1` are converted to the new `v1` API when read in Kubernetes 1.22.
```

{{% /notice %}}

> ##### Verify upgrade

Reviewer: ```suggestion
## Verify upgrade
```
